File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int n = heights.size();
        int i =0, j= n-1, k =0;
        int res=0;

        while(i<j){
            int height = min(heights[i], heights[j]);
            int len = j-i;
            res = max(res, height*len);
            (heights[i]>=heights[j]) ? j-- : i++;
            // k =j;
            // while(k>i){
            //     int height = min(heights[i], heights[k]);
            //     int len = k-i;
            //     cout << height << len << endl;
            //     res = max(res, height*len);
            //     cout << res << endl;
            //     k--;
            // }
            // i++;
        }

        return res;
        
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l=1
        r=max(piles)
        res = float('inf')

        while l<=r:
            mid = (l+r)//2
            max_ba=0
            for n in piles:
                max_ba += math.ceil(n/mid)
            if max_ba > h:
                l=mid+1
            else:
                res = mid
                r=mid-1
        
        return res